File: omnibook-broker/src/main/java/com/sprint/omnibook/broker/domain/repository/ReservationEventRepository.java
package com.sprint.omnibook.broker.domain.repository;

import com.sprint.omnibook.broker.domain.ReservationEventEntity;
import com.sprint.omnibook.broker.event.PlatformType;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface ReservationEventRepository extends JpaRepository<ReservationEventEntity, Long> {

    /**
     * 이벤트 ID로 조회.
     */
    Optional<ReservationEventEntity> findByEventId(UUID eventId);

    /**
     * 미처리 이벤트 조회.
     */
    List<ReservationEventEntity> findByProcessedFalseOrderByReceivedAtAsc();

    /**
     * 플랫폼과 예약 ID로 이벤트 조회.
     */
    List<ReservationEventEntity> findByPlatformTypeAndPlatformReservationIdOrderByReceivedAtDesc(
            PlatformType platformType, String platformReservationId);
}

File: omnibook-broker/src/main/kotlin/com/sprint/omnibook/broker/domain/repository/ReservationEventRepository.kt
package com.sprint.omnibook.broker.domain.repository

import com.sprint.omnibook.broker.domain.ReservationEventEntity
import com.sprint.omnibook.broker.event.PlatformType
import org.springframework.data.jpa.repository.JpaRepository
import java.util.UUID

interface ReservationEventRepository : JpaRepository<ReservationEventEntity, Long> {

    /**
     * 이벤트 ID로 조회.
     */
    fun findByEventId(eventId: UUID): ReservationEventEntity?

    /**
     * 미처리 이벤트 조회.
     */
    fun findByProcessedFalseOrderByReceivedAtAsc(): List<ReservationEventEntity>

    /**
     * 플랫폼과 예약 ID로 이벤트 조회.
     */
    fun findByPlatformTypeAndPlatformReservationIdOrderByReceivedAtDesc(
        platformType: PlatformType,
        platformReservationId: String
    ): List<ReservationEventEntity>
}
